### PR TITLE
docs: added documentation typedoc comments for icon component

### PR DESCRIPTION
**Pull request by InnaAtanasova**
#### Please provide a link to the associated issue.

part of [#687](https://github.com/SAP/fundamental-ngx/issues/687)

#### Please provide a brief summary of this pull request.

added documentation typedoc comments for icon component

### CODE REVIEW

File: library/src/lib/icon/icon.directive.ts [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
+ * The base class for the icon component 
+ */
 const BASE_ICON_CLASS = 'sap-icon';
 
-/** Prefix for icon prop classes */
+/** 
+ * @hidden
+ * Prefix for icon prop classes 
+ */
+
```
Comment (MattL75) [inline library/src/lib/icon/icon.directive.ts]: remove extra line

File: library/src/lib/icon/icon.directive.ts [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
 const PREFIX_ICON_CLASS = BASE_ICON_CLASS + '--';
 
 @Directive({
```
Comment (MattL75) [inline library/src/lib/icon/icon.directive.ts]: document class

File: library/src/lib/icon/icon.directive.ts [hunk 3]
```diff
@@ -30,6 +45,7 @@ export class IconDirective extends AbstractFdNgxClass {
         }
     }
 
+    /** @hidden */
     constructor(@Inject(ElementRef) elementRef: ElementRef) {
```
Comment (MattL75) [inline library/src/lib/icon/icon.directive.ts]: can you remove the @Inject(ElementRef) and replace with just 'private' as part of this pr pls?

File: library/src/lib/icon/icon.directive.ts (hunk 2 above)
Comment (InnaAtanasova) [reply library/src/lib/icon/icon.directive.ts]: done

File: library/src/lib/icon/icon.directive.ts (hunk 1 above)
Comment (InnaAtanasova) [reply library/src/lib/icon/icon.directive.ts]: done

File: library/src/lib/icon/icon.directive.ts (hunk 3 above)
Comment (InnaAtanasova) [reply library/src/lib/icon/icon.directive.ts]: done

**Pull request merged**